### DISCUSSION TITLE
Add binary sensor device class bypassed?

### DISCUSSION THREAD

**Discussion by rlippmann** [home-assistant/architecture#1048] [Entity Models]
Hi all,

I was looking to add a binary sensor entity to a sensor device for when the sensor is bypassed.

My use case is for an alarm system where a faulted/tripped zone can be bypassed to arm an alarm, but I imagine there are other use cases.  

Looking at the device categories, the closest is probably connectivity, but that's not quite it.

Making the device unavailable isn't really a solution, because the device state can still change.  And I know that setting an attribute on the device is generally not the way one is supposed to do things like this.

So, I'm proposing that a bypassed device class be added.

Thoughts?

**Comment by gjohansson-ST** [comment]
It sounds very specific with very limited use-case.
May I suggest to use the `problem` device class instead or perhaps even no device class and dynamic icons instead (as I guess that's what you're mostly after)?

**Reply by rlippmann** [reply]
Well, it's not technically a problem, but no device class with dynamic icons is probably a better way.  I was just wondering if it wasn't a specific use case.

Perhaps I should open another issue for this, but maybe BinarySensorEntityDescription should have an optional 2nd icon for cases such as this?

**Reply by joostlek** [reply]
You mean a second icon for when it's turned on? You can use icon translations for that now

**Reply by parautenbach** [reply]
I only see the recently introduced state translations. Where are these icon translations you're referring to? Or are you simply referring to icon templates using if statements?

https://www.home-assistant.io/docs/configuration/templating/

**Reply by gjohansson-ST** [reply]
See https://developers.home-assistant.io/blog/2024/01/19/icon-translations/
But having different icons for different states is not something new and the integration proving the entity can set it's own icons instead of using device class if wanted and the existing device classes are not suitable.
